src/cells/model.ts

```diff
@@ -429,6 +438,15 @@ class CodeCellModel extends BaseCellModel implements ICodeCellModel {
   }
 
   /**
+   * Clear the cell state.
+   */
+  clear(): void {
+    this.output.clear(false);
+    this.executionCount = null;
+    this.input.prompt = 'In [ ]:';
```

Comment:
That makes sense too. Right now the cell seems to be micro-managing the input prompt in a lot of places, so I'm okay leaving this as-is now and coming back to it later.